### PR TITLE
Add message handling for block registry

### PR DESCRIPTION
**Pull request by drslebedev**
Add message handling for registry block types and registry known block parametrizations.

Related issue: https://github.com/fair-acc/opendigitizer/issues/247

### CODE REVIEW

**Comment by ivan-cukic** [comment]
To increase usability, can you return both the types and parametrizations in the same message? (to avoid having state machines in the message sender and to avoid the roundtrips)

Something like:
```
{
    { "type", "blockType1" }
    { "parametrizations", { "float,float", "float,double" }
},
{
    { "type", "blockType2" }
    { "parametrizations", { "float", "double" }
}
```

File: core/include/gnuradio-4.0/Graph.hpp [hunk 1]
```diff
@@ -532,6 +541,20 @@ class Graph : public gr::Block<Graph> {
...
+        assert(propertyName == graph::property::kRegistryBlockTypes);
+        PluginLoader&                   loader      = gr::globalPluginLoader();
+        const std::vector<std::string>& knownBlocks = loader.knownBlocks();
+
+        property_map result{};
+        for (const auto& blockType : knownBlocks) {
+            std::vector<std::string_view> knownBlockParams = loader.knownBlockParameterizations(blockType);
+            result[blockType]                              = property_map{{"parametrizations", std::vector<std::string>(knownBlockParams.begin(), knownBlockParams.end())}};
+        }
+        message.data = property_map{{"types", result}};
```
Comment (ivan-cukic) [inline core/include/gnuradio-4.0/Graph.hpp:554]: Could be `std::move(result)`

File: core/include/gnuradio-4.0/Graph.hpp [hunk 2]
```diff
@@ -880,10 +903,10 @@ class MergedGraph<Left, Right, OutId, InId> : public Block<MergedGraph<Left, Rig
             auto right_out = apply_right<InId, traits::block::stream_input_port_types<Right>::size() - InId - 1>(std::forward_as_tuple(std::forward<Ts>(inputs)...), std::move(std::get<OutId>(left_out)));
 
             if constexpr (traits::block::stream_output_port_types<Left>::size == 2 && traits::block::stream_output_port_types<Right>::size == 1) {
-                return std::make_tuple(std::move(std::get<OutId ^ 1>(left_out)), std::move(right_out));
+                return std::make_tuple(std::move(std::get < OutId ^ 1 > (left_out)), std::move(right_out));
```
Comment (ivan-cukic) [inline core/include/gnuradio-4.0/Graph.hpp:906]: Looks like clang-format got confused :)

**Comment by ivan-cukic** [comment]
Other than the above nitpicks, approved

File: core/include/gnuradio-4.0/Graph.hpp (hunk 1 above)
Comment (drslebedev) [reply core/include/gnuradio-4.0/Graph.hpp]: Done

**Review by ivan-cukic — APPROVED** [review]

**Pull request merged**